apm-agent-plugins/apm-apache-httpclient/apm-apache-httpclient4-plugin/src/main/java/co/elastic/apm/agent/httpclient/v4/ApacheHttpAsyncClientInstrumentation.java

```diff
@@ -89,41 +91,44 @@ public static class ApacheHttpAsyncClientAdvice {
         public static Object[] onBeforeExecute(@Advice.Argument(value = 0) HttpAsyncRequestProducer requestProducer,
                                                @Advice.Argument(2) HttpContext context,
                                                @Advice.Argument(value = 3) FutureCallback<?> futureCallback) {
-            AbstractSpan<?> parent = tracer.getActive();
-            if (parent == null) {
+
+            ElasticContext<?> parentContext = tracer.currentContext();
+            if (parentContext.isEmpty()) {
+                // performance optimization, no need to wrap if we have nothing to propagate
+                // empty context means also we will not create an exit span
                 return null;
             }
-            Span<?> span = parent.createExitSpan();
-            HttpAsyncRequestProducer wrappedProducer = requestProducer;
             FutureCallback<?> wrappedFutureCallback = futureCallback;
-            boolean responseFutureWrapped = false;
-            if (span != null) {
-                span.withType(HttpClientHelper.EXTERNAL_TYPE)
-                    .withSubtype(HttpClientHelper.HTTP_SUBTYPE)
-                    .withSync(false)
-                    .activate();
-
-                wrappedProducer = asyncHelper.wrapRequestProducer(requestProducer, span, null);
-                wrappedFutureCallback = asyncHelper.wrapFutureCallback(futureCallback, context, span);
-                responseFutureWrapped = true;
-            } else {
-                wrappedProducer = asyncHelper.wrapRequestProducer(requestProducer, null, parent);
+            Span<?> span = null;
+            ElasticContext<?> parent = tracer.currentContext();
```

Comment:
[minor] do you think that would make more sense to call this kind of variables `activeContext` instead of `parent` ? Since the context can now be something else than a parent in a parent/child relationship I think it would make that a bit more clear.

Reply:
I'm fine with both. The `ElasticContext` is now the one starting the span, so technically it is the parent. The created span inherits the parent-span and the baggage from it's parent context.
I'll rename it to `activeContext` though